drivers/storage/portworx/components_test.go

```diff
@@ -2794,6 +2803,128 @@ func TestSecurityInstall(t *testing.T) {
 	err = testutil.Get(k8sClient, systemSecret, pxutil.SecurityPXSystemSecretsSecretName, cluster.Namespace)
 	require.NoError(t, err)
 	require.Equal(t, systemSecretKey, systemSecret.Data[pxutil.SecuritySystemSecretKey])
+
+}
+
+func TestGuestAccessSecurity(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	// Create the mock servers that can be used to mock SDK calls
+	mockRoleServer := mock.NewMockOpenStorageRoleServer(mockCtrl)
+	mockClusterServer := mock.NewMockOpenStorageClusterServer(mockCtrl)
+	mockNodeServer := mock.NewMockOpenStorageNodeServer(mockCtrl)
+
+	// Start a sdk server that implements the mock servers
+	sdkServerIP := "127.0.0.1"
+	sdkServerPort := 23888
+	mockSdk := mock.NewSdkServer(mock.SdkServers{
+		Role:    mockRoleServer,
+		Node:    mockNodeServer,
+		Cluster: mockClusterServer,
+	})
+	err := mockSdk.StartOnAddress(sdkServerIP, strconv.Itoa(sdkServerPort))
+	require.NoError(t, err)
+	defer mockSdk.Stop()
+
+	k8sClient := testutil.FakeK8sClient(&v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pxutil.PortworxServiceName,
+			Namespace: "kube-test",
+		},
+		Spec: v1.ServiceSpec{
+			ClusterIP: sdkServerIP,
+			Ports: []v1.ServicePort{
+				{
+					Name: pxutil.PortworxSDKPortName,
+					Port: int32(sdkServerPort),
+				},
+			},
+		},
+	})
+	driver := portworx{}
+	driver.Init(k8sClient, runtime.NewScheme(), record.NewFakeRecorder(10))
+	cluster := &corev1alpha1.StorageCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "px-cluster",
+			Namespace: "kube-test",
+		},
+		Spec: corev1alpha1.StorageClusterSpec{
+			Security: &corev1alpha1.SecuritySpec{
+				Enabled: true,
+				Auth: &corev1alpha1.AuthSpec{
+					GuestAccess: &corev1alpha1.GuestRoleManaged,
+					SelfSigned: &corev1alpha1.SelfSignedSpec{
+						TokenLifetime: &metav1.Duration{
+							// Since we have a token expiration buffer of one minute,
+							// a new token will constantly be fetched.
+							Duration: time.Second * 1,
+						},
+					},
+				},
+			},
+		},
+	}
+	// Initial run
+	setSecuritySpecDefaults(cluster)
+	securityComponent, ok := component.Get(component.SecurityComponentName)
+	require.True(t, ok)
+	component.DeregisterAllComponents()
+	component.Register(component.SecurityComponentName, securityComponent)
+	securityComponent.Initialize(k8sClient, hversion.Version{}, runtime.NewScheme(), record.NewFakeRecorder(10))
+	driver.UpdateStorageClusterStatus(cluster)
+	require.NoError(t, err)
```

Comment:
Do we need any of this? Why do we need to call UpdateStorageClusterStatus? I guess, if we don't call that, we don't even need Node and Cluster services.

Reply:
Yeah, the test was failing with other components running. I think due to the mock server. I just wanted to isolate this test to only run the security operator.

Reply:
Just removed the node/cluster servers as we don't need them.